fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PolymorphicFunctionAnalyzer.java

```diff
@@ -209,8 +211,14 @@ private static Function resolveByDeducingReturnType(Function fn, Type[] inputArg
         if (deduce == null) {
             return null;
         }
-        Type[] resolvedArgTypes = resolveArgTypes(fn, inputArgTypes);
+        // Apply deduce logic first,then replace null to boolean
+        // because null type can have common super type with all types but boolean can't
+        Type[] resolvedArgTypes = Arrays.copyOf(inputArgTypes, inputArgTypes.length);
         Type newRetType = deduce.apply(resolvedArgTypes);
```

Comment:
Could these logic be put inside the function `resolveArgTypes`?

Reply:
> Could these logic be put inside the function `resolveArgTypes`?

make resolveArgTypes' logic more clear